core/CronArchive.php

```diff
@@ -262,6 +267,8 @@ public function __construct($processNewSegmentsFrom = null, LoggerInterface $log
 
         $processNewSegmentsFrom = $processNewSegmentsFrom ?: StaticContainer::get('ini.General.process_new_segments_from');
         $this->segmentArchivingRequestUrlProvider = new SegmentArchivingRequestUrlProvider($processNewSegmentsFrom);
+
+        $this->invalidator = StaticContainer::get('Piwik\Archive\ArchiveInvalidator');
```

Comment:
Was wondering why this is here and not in the method `invalidateArchivedReportsForSitesThatNeedToBeArchivedAgain` but I guess this is to prepare later to remove the StaticContainer later easier? Should we try to do this usually? I usually do this in the method itself but if it's useful I will try to move to `__constrcut` when it makes sense